File: model/src/main/java/net/artux/pda/model/quest/MissionModel.java
package net.artux.pda.model.quest;

import java.io.Serializable;
import java.util.List;

import lombok.Data;

@Data
public class MissionModel implements Serializable {
    private String title;
    private String name;
    private List<CheckpointModel> checkpoints;

    public boolean hasParams(String... params) {
        return getCurrentCheckpoint(params) != null;
    }

    public CheckpointModel getCurrentCheckpoint(String... params) {
        for (CheckpointModel checkpointModel : checkpoints) {
            if (checkpointModel.isActual(params))
                return checkpointModel;
        }
        return null;
    }

}

File: model/src/main/java/net/artux/pda/model/quest/MissionModel.kt
package net.artux.pda.model.quest

import java.io.Serializable

class MissionModel : Serializable {
    var title: String? = null
    var name: String? = null
    var checkpoints: List<CheckpointModel>? = null

    fun hasParams(vararg params: String): Boolean {
        return getCurrentCheckpoint(*params) != null
    }

    fun getCurrentCheckpoint(vararg params: String): CheckpointModel? {
        for (checkpointModel in checkpoints!!) {
            if (checkpointModel.isActual(*params)) return checkpointModel
        }
        return null
    }
}